local.tf

```diff
@@ -80,6 +80,7 @@ locals {
     spot_allocation_strategy                 = "lowest-price"                                   # The only valid value is lowest-price, which is also the default value. The Auto Scaling group selects the cheapest Spot pools and evenly allocates your Spot capacity across the number of Spot pools that you specify.
     spot_instance_pools                      = 10                                               # "Number of Spot pools per availability zone to allocate capacity. EC2 Auto Scaling selects the cheapest Spot pools and evenly allocates Spot capacity across the number of Spot pools that you specify."
     spot_max_price                           = ""                                               # Maximum price per unit hour that the user is willing to pay for the Spot instances. Default is the on-demand price
+    use_spot_instances                       = false                                            # set to true to use spot instances with a LAUNCH TEMPLATE
```

Comment:
Can we just have `null` here and then no conditional? i.e.

```
market_type = lookup(
      var.worker_groups_launch_template[count.index],
      "use_spot_instances",
    local.workers_group_defaults["use_spot_instances"])
  }
```

Reply:
I don't exactly get why... I just wanted a simple flag it to tell it to use spot in the launch template?

Reply:
Because now you have `value` > `conditional` > `other value`.

e.g. `true/false` > `conditional` > `string(spot)/null`

But it's simpler, and more like most of the other attributes, to just have the value itself with a default of `""`. See what I mean?

Reply:
so... a `market_type = "spot" | ""` instead? Less code change but less user friendly? Telling the user to leave it as `""` to use on-demand and use `"spot"` for spot instances, did I catch that right? Sorry I'm too rookie with Terraform, hence all the questions.

Reply:
Yes, exactly that.